### PR TITLE
upgrade JSONField for product data

### PR DESCRIPTION
**Pull request by nickpell**
upgrading our third party JSONField library from 0.5.7 to 1.0.3 (same workflow as https://github.com/dimagi/commcare-hq/pull/10432)

### CODE REVIEW

File: corehq/apps/products/migrations/0001_initial.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ class Migration(migrations.Migration):
                 ('program_id', models.CharField(default=b'', max_length=100, null=True)),
                 ('cost', models.DecimalField(null=True, max_digits=20, decimal_places=5)),
                 ('units', models.CharField(default=b'', max_length=100, null=True)),
-                ('product_data', json_field.fields.JSONField(default={}, help_text='Enter a valid JSON object')),
+                ('product_data', jsonfield.fields.JSONField(default={})),
```
Comment (nickpell) [inline corehq/apps/products/migrations/0001_initial.py]: This removes the old default ```help_text``` in ```json_field```.  Switching to ```jsonfield``` would require adding a new migration with an ```AlterField``` to get rid of the ```help_text```.  I choose just to remove it here; I'm not aware of any negative side-effects.

Comment (nickpell) [inline corehq/apps/products/migrations/0001_initial.py]: http://stackoverflow.com/questions/26503826/why-does-django-generate-migrations-for-help-text-and-verbose-name-changes?answertab=votes#tab-top

**Comment by czue** [comment]
@nickpell do these need any testing at all?

**Comment by nickpell** [comment]
This worked earlier: https://github.com/dimagi/commcare-hq/pull/10432

Since that was ```list``` data and this defaults to ```dict``` data, I tested saving ```product_data``` using ```json_field``` and making sure it was available and editable using ```jsonfield``` (and vice-versa) and everything worked as expected.  So I'm happy with the level of testing.

File: corehq/apps/products/migrations/0001_initial.py (hunk 1 above)
Comment (emord) [inline corehq/apps/products/migrations/0001_initial.py]: Should this be changed to `dict` as well?

Comment (nickpell) [inline corehq/apps/products/migrations/0001_initial.py]: jinx https://github.com/dimagi/commcare-hq/pull/11880

**Pull request merged**